flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/util/HiveTypeUtil.java

```diff
@@ -276,4 +198,202 @@ private static DataType toFlinkPrimitiveType(PrimitiveTypeInfo hiveType) {
+		}
+
+		@Override
+		public TypeInfo visit(TinyIntType tinyIntType) {
+			return TypeInfoFactory.byteTypeInfo;
+		}
+
+		@Override
+		public TypeInfo visit(SmallIntType smallIntType) {
+			return TypeInfoFactory.shortTypeInfo;
+		}
+
+		@Override
+		public TypeInfo visit(IntType intType) {
+			return TypeInfoFactory.intTypeInfo;
+		}
+
+		@Override
+		public TypeInfo visit(BigIntType bigIntType) {
+			return TypeInfoFactory.longTypeInfo;
+		}
+
+		@Override
+		public TypeInfo visit(FloatType floatType) {
+			return TypeInfoFactory.floatTypeInfo;
+		}
+
+		@Override
+		public TypeInfo visit(DoubleType doubleType) {
+			return TypeInfoFactory.doubleTypeInfo;
+		}
+
+		@Override
+		public TypeInfo visit(DateType dateType) {
+			return TypeInfoFactory.dateTypeInfo;
+		}
+
+		@Override
+		public TypeInfo visit(TimeType timeType) {
+			return null;
+		}
+
+		@Override
+		public TypeInfo visit(TimestampType timestampType) {
+			return TypeInfoFactory.timestampTypeInfo;
+		}
+
+		@Override
+		public TypeInfo visit(ZonedTimestampType zonedTimestampType) {
+			return null;
+		}
+
+		@Override
+		public TypeInfo visit(LocalZonedTimestampType localZonedTimestampType) {
+			return null;
+		}
+
+		@Override
+		public TypeInfo visit(YearMonthIntervalType yearMonthIntervalType) {
+			return null;
+		}
+
+		@Override
+		public TypeInfo visit(DayTimeIntervalType dayTimeIntervalType) {
+			return null;
+		}
+
+		@Override
+		public TypeInfo visit(ArrayType arrayType) {
+			LogicalType elementType = arrayType.getElementType();
+			TypeInfo elementTypeInfo = elementType.accept(new TypeInfoLogicalTypeVisitor());
+			if (null != elementTypeInfo) {
+				return TypeInfoFactory.getListTypeInfo(elementTypeInfo);
+			} else {
+				return null;
+			}
+		}
+
+		@Override
+		public TypeInfo visit(MultisetType multisetType) {
+			return null;
+		}
+
+		@Override
+		public TypeInfo visit(MapType mapType) {
+			LogicalType keyType  = mapType.getKeyType();
+			LogicalType valueType = mapType.getValueType();
+			TypeInfo keyTypeInfo = keyType.accept(new TypeInfoLogicalTypeVisitor());
+			TypeInfo valueTypeInfo = valueType.accept(new TypeInfoLogicalTypeVisitor());
+			if (null == keyTypeInfo || null == valueTypeInfo) {
+				return null;
+			} else {
+				return TypeInfoFactory.getMapTypeInfo(keyTypeInfo, valueTypeInfo);
+			}
+		}
+
+		@Override
+		public TypeInfo visit(RowType rowType) {
+			List<String> names = rowType.getFieldNames();
+			List<TypeInfo> typeInfos = new ArrayList<>(names.size());
+			for (String name : names) {
+				TypeInfo typeInfo =
+						rowType.getTypeAt(rowType.getFieldIndex(name)).accept(new TypeInfoLogicalTypeVisitor());
+				if (null != typeInfo) {
+					typeInfos.add(typeInfo);
+				} else {
+					return null;
+				}
+			}
+			return TypeInfoFactory.getStructTypeInfo(names, typeInfos);
+		}
+
+		@Override
+		public TypeInfo visit(DistinctType distinctType) {
```

Comment:
Some last suggestion: Instead of returning null a dozen of times you could simply use the provided `LogicalTypeDefaultVisitor` and throw the exception in the `defaultMethod` there.

Reply:
That would be really cool!
Thanks for your feedback.